pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -351,19 +352,19 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 			waitingPod.Reject(pluginName, "preempted")
 		} else {
 			if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
-				condition := &v1.PodCondition{
-					Type:    v1.AlphaNoCompatGuaranteeDisruptionTarget,
-					Status:  v1.ConditionTrue,
-					Reason:  "PreemptionByKubeScheduler",
-					Message: "Kube-scheduler: preempting",
-				}
-				newStatus := pod.Status.DeepCopy()
-				if apipod.UpdatePodCondition(newStatus, condition) {
-					if err := util.PatchPodStatus(ctx, cs, victim, newStatus); err != nil {
-						klog.ErrorS(err, "Preparing pod preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
-						errCh.SendErrorWithCancel(err, cancel)
-						return
-					}
+				victimPodApply := corev1apply.Pod(victim.Name, victim.Namespace).WithStatus(corev1apply.PodStatus())
+				victimPodApply.Status.WithConditions(corev1apply.PodCondition().
+					WithType(v1.AlphaNoCompatGuaranteeDisruptionTarget).
+					WithStatus(v1.ConditionTrue).
+					WithReason("PreemptionByKubeScheduler").
+					WithMessage("Kube-scheduler: preempting").
+					WithLastTransitionTime(metav1.Now()),
+				)
+
+				if _, err := cs.CoreV1().Pods(victim.Namespace).ApplyStatus(ctx, victimPodApply, metav1.ApplyOptions{FieldManager: "Scheduler", Force: true}); err != nil {
```

Comment:
is there a rule that the field manager is camel case? We should be using the profile name (which matches the scheduler name)

Reply:
I don't think there is a rule it should be camel case or in any other format, the documentation of the field is pretty relaxed on this: https://github.com/kubernetes/kubernetes/blob/5539a5b80fe45cc04df4f53bd1047cd6d998bcf1/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L541. In this blog post hypen is used: https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/. I propose camel case just to pick something and be consistent as being consistent was one of the remarks by @apelisse. Also, we generally use camel-case in the `reason` field when naming the actor, for example `PreemptionByKubeScheduler`.

As for the profile name - could you explain what the value is, is it a constant? 

I searched for profile name within the package and found this "https://github.com/kubernetes/kubernetes/blob/5539a5b80fe45cc04df4f53bd1047cd6d998bcf1/pkg/scheduler/framework/interface.go#L592, but it doesn't seem ready to use in `preemption.go` as it is not passed via `PostFilter` invocations. I guess it would require extending the set of parameters when calling `PostFilter` or passing the value in context as a keyed value. 

I think staying with a constant value for the actor is good enough, until we have a scenario in which it is problematic.